app/grandchallenge/workstations/models.py

```diff
@@ -412,13 +454,16 @@ def environment(self) -> dict:
         -------
             The environment variables that should be set on the container.
         """
-        env = {
-            "NGINX_RESOLVER": settings.WORKSTATIONS_DNS_RESOLVER,
-            "GRAND_CHALLENGE_API_ROOT": unquote(reverse("api:api-root")),
-            "WORKSTATION_SENTRY_DSN": settings.WORKSTATION_SENTRY_DSN,
-            "WORKSTATION_SESSION_ID": str(self.pk),
-            "CIRRUS_KEEP_ALIVE_METHOD": "old",
-        }
+        env = {var["name"]: var["value"] for var in self.extra_env_vars}
+
+        env.update(
```

Comment:
You could also additionally prevent those from being set in the schema as well. Still need to be overwritten of course, but it would make it clear that those values cannot be user-defined.

Reply:
Sure, but then you would need to maintain the list of all the un-settable ones in the Regex of the name value validator. I don't think that we'd ever do that, and no big deal if a user tries to set that and it doesn't work.